### ISSUE TITLE
Object/Item Development

### ISSUE THREAD

**Issue by lindsey-hsi** [uchicago-cs/chiventure#52]
In this task, we aim to further develop the objects/items module within game-state. This will include implementing an improved object data structure with a hash table storing attributes (to replace the tagged union), working/fleshing out accompanying functions, and accommodating any relevant requests from other teams.

**Comment by borjasotomayor** [comment]
I took a look at this code (since it came up in #60), and it seems like trying to support attribute values that can be either char, double, int, or char* is excessive (https://github.com/uchicago-cs/chiventure/blob/317f28535aa031dd050f1e34e1af299402dce169/src/game-state/include/object.h#L11-L17) Why not constraint it just to strings and ints?

Also, not sure I understand this comment above the definition of attribute values:

```
//actual values left up to action management
```

While action management may affect attribute values, those values will also be loaded when the game is constructed by WDL.

**Comment by lindsey-hsi** [comment]
Closing Issue Overview:
It should be noted that we changed all "object" names to "item" in order to prevent confusion regarding objects amongst other teams. In this sprint, we altered the item data structure to include a hash table of possible attributes that an item may possess (i.e. whether or not a door is locked, if an item can be taken by a player, etc). This hash table holds string keys which describe characteristics of the item, and values that support a variety of data types (specifically those that are supported by the YAML parser as suggested by Borja). Further, we wrote implementations for the functions previously outlined in Sprint 1, and defined + implemented accompanying functions that interact with the new attribute table. In this coming sprint, we plan on further developing the attributes aspect of item alongside writing rigorous tests to examine the functionality of our work thus far. In addition, we hope to work very closely with other teams in effort to integrate our item framework with their needs, towards the goal of creating a basic prototype at the end of the next two weeks.
